cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -180,33 +182,49 @@ func (mgr *packetManagerRest) listPacketDevices() (*Devices, error) {
 	var jsonStr = []byte(``)
 	packetAuthToken := os.Getenv("PACKET_AUTH_TOKEN")
 	url := mgr.baseURL + "/projects/" + mgr.projectID + "/devices"
-	req, _ := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+
+	req, err := http.NewRequest("GET", url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list Packet devices: %w", err)
```

Comment:
The function name is called `listPacketDevices()`, so `failed to list Packet devices` error does not provide any information to the caller. That also helps with tracking failing codepath.
```suggestion
		return nil, fmt.Errorf("failed to create request: %w", err)
```